cpp/src/arrow/acero/asof_join_node.cc

```diff
@@ -668,13 +668,13 @@ class InputState {
 
   static Result<std::unique_ptr<InputState>> Make(
       size_t index, TolType tolerance, bool must_hash, bool may_rehash,
-      KeyHasher* key_hasher, AsofJoinNode* node, ExecNode* output,
+      KeyHasher* key_hasher, AsofJoinNode* node, ExecNode* input,
```

Comment:
This still looking confusing:

(1) BackpressureController takes `ExecNode* node, ExecNode* output` and this one nows takes `AsofJoinNode* node, ExecNode* input` which is inconsistent

(2) InputState::Make uses to take inputs[i] but now it takes "node" which is the AsofJoinNode

Reply:
It is also confusing that 

```
    return std::make_unique<InputState>(index, tolerance, must_hash, may_rehash,
                                        key_hasher, node, std::move(handler), schema,
                                        time_col_index, key_col_index);
```
On line 681 passes the asof join node to the input state instead of the input node, why is it?

Reply:
@rtpsw I think it is easier to revert this line as to unblock the internal first:
https://github.com/apache/arrow/pull/34392/files#diff-5493b6ae7ea2a4d5cfb581034c076e9c4be7608382168de6d1301ef67b6c01eeR1410

Then work on cleaning up changes introduced in GH-36391. The code is quite confusing now.

Reply:
> I think it is easier to revert this line

This reversion alone doesn't compile because `inputs[i]` is not of type `AsofJoinNode*`.

Reply:
It's probably easier for me to do it instead of back and forth, opened:

https://github.com/apache/arrow/pull/35878

Reply:
I am fine with figure out the cleaner way to do this in follow up to GH-36391. But for now I think it's easier to just revert the change.

Reply:
> The two places have a different perspective of what is an input and what is an output (which likely caused confusion in the first place).

We should probably fix the variable naming in the follow PR to GH-36391 how to call these things then. But for now let's just revert to what was before.

Reply:
Maybe use more specific names like `asof_input` and `asof_node`?

Reply:
Done.